src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -306,9 +306,9 @@ public static void EndInvokeJS(JSRuntime jsRuntime, string arguments)
         var success = reader.GetBoolean();
 
         reader.Read();
-        jsRuntime.EndInvokeJS(taskId, success, ref reader);
+        var completed = jsRuntime.EndInvokeJS(taskId, success, ref reader);
```

Comment:
Any reason we wouldn't want to do? I imagine we don't care about reading more JSON if it previously failed?

```suggestion
        if (!jsRuntime.EndInvokeJS(taskId, success, ref reader))
        {
           return;
        }
```